app/Http/Controllers/Admin/AdminReportController.php

```diff
@@ -1327,4 +1349,173 @@ public function reportsApiSpamGet(Request $request, $id)
 
         return new AdminSpamReport($report);
     }
+
+    public function reportsApiRemoteHandle(Request $request)
+    {
+        $this->validate($request, [
+            'id' => 'required|exists:remote_reports,id',
+            'action' => 'required|in:mark-read,cw-posts,unlist-posts,delete-posts,private-posts,mark-all-read-by-domain,mark-all-read-by-username,cw-all-posts,private-all-posts,unlist-all-posts'
+        ]);
+
+        $report = RemoteReport::findOrFail($request->input('id'));
+        $user = User::whereProfileId($report->account_id)->first();
+        $ogPublicStatuses = [];
+        $ogUnlistedStatuses = [];
+        $ogNonCwStatuses = [];
+
+        switch ($request->input('action')) {
+            case 'mark-read':
+                $report->action_taken_at = now();
+                $report->save();
+                break;
+            case 'mark-all-read-by-domain':
+                RemoteReport::whereInstanceId($report->instance_id)->update(['action_taken_at' => now()]);
+                break;
+            case 'cw-posts':
+                $statuses = Status::find($report->status_ids);
+                foreach($statuses as $status) {
+                    if($report->account_id != $status->profile_id) {
```

Comment:
This filter should happen on the receiving side I think? Keep in mind Mastodon will send you Flag activities where your user is not the reported user, but rather a user mentioned in that reply.